ensure_tiller_installed.go

```diff
@@ -35,6 +36,9 @@ func (c *Client) EnsureTillerInstalled(ctx context.Context) error {
 func (c *Client) EnsureTillerInstalledWithValues(ctx context.Context, values []string) error {
 	// Check if Tiller is already present and return early if so.
 	{
+		ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		defer cancel()
```

Comment:
We do this in the migration resources in cluster-operator and it works well. Giving it 10 secs because of the dumb port forwarding connection.